### PR TITLE
More auth fixes

### PR DESCRIPTION
**Pull request by neb42**
This fixes an issue where all tokens had expired and the UI wouldn't fallback to the login screen

### CODE REVIEW

File: frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts [hunk 1]
```diff
@@ -27,7 +27,11 @@ const getStoredTokenResponse = (): TokenResponse | undefined => {
 const storeTokenResponse = (tokenResponse: TokenResponse) => {
   try {
     if (Platform.OS !== 'web') return;
-    sessionStorage.setItem('tokenResponse', JSON.stringify(tokenResponse));
+    if (tokenResponse) {
```
Comment (internetti) [inline frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts:30]: might be useful to write a test for this

File: frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts [hunk 2]
```diff
@@ -131,7 +135,11 @@ export const useTokenResponse = (): [
       if (refreshTokenInterval.current)
         clearInterval(refreshTokenInterval.current);
     };
-  }, [tokenResponse?.refreshToken, tokenResponse?.expiresIn]);
+  }, [
+    tokenResponse?.refreshToken,
+    tokenResponse?.expiresIn,
+    JSON.stringify(discovery),
```
Comment (ludydoo) [inline frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts:141]: That might be expensive to stringify for change detection. Would this work with `discovery` alone ?

Comment (neb42) [reply frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts:141]: Ran a test for it and it's below a ms

File: frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts (hunk 1 above)
Comment (internetti) [reply frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts:30]: @neb42 are you planning to write a test here, or do you consider this PR done?

Comment (neb42) [reply frontend/apps/core-app/src/components/AuthWrapper/useTokenResponse.ts:30]: Having trouble mocking session storage. I think the logic here is simple enough to leave it

**Pull request merged**